File: ProjectPUM/app/src/main/java/com/example/mycalendar/MainActivity.java
package com.example.mycalendar;

import static com.example.mycalendar.util.CalendarUtils.daysInMonthArray;
import static com.example.mycalendar.util.CalendarUtils.monthYearDate;
import static com.example.mycalendar.util.CalendarUtils.selectedDate;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;
import com.example.mycalendar.adapter.CalendarAdapter;
import java.time.LocalDate;
import java.util.ArrayList;

public class MainActivity extends AppCompatActivity implements CalendarAdapter.OnItemListener {

    private TextView monthYearText;
    private RecyclerView calendarRecyclerView;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        initWidgets();
        selectedDate = LocalDate.now();
        setMonthView();

    }

    private void initWidgets() {

        calendarRecyclerView = findViewById(R.id.calendarRecyclerView);
        monthYearText = findViewById(R.id.monthYearTextView);

    }

    private void setMonthView() {

        monthYearText.setText(monthYearDate(selectedDate));
        ArrayList<LocalDate> daysInMonth = daysInMonthArray(selectedDate);

        CalendarAdapter calendarAdapter = new CalendarAdapter(daysInMonth, this);
        RecyclerView.LayoutManager layoutManager = new GridLayoutManager(getApplicationContext(), 7);
        calendarRecyclerView.setLayoutManager(layoutManager);
        calendarRecyclerView.setAdapter(calendarAdapter);

    }

    public void nextMonthAction(View view) {
        selectedDate = selectedDate.plusMonths(1);
        setMonthView();
    }

    public void previousMonthAction(View view) {
        selectedDate = selectedDate.minusMonths(1);
        setMonthView();
    }

    @Override
    public void onItemClick(int position, LocalDate date) {

        if(date != null) {
            selectedDate = date;
            setMonthView();
        }

    }

    public void weeklyAction(View view) {
        startActivity(new Intent(this, WeeklyCalendarActivity.class));
    }
}

File: FizQuiz/app/src/main/java/com/example/fizquiz/MainActivity.kt
package com.example.fizquiz

import android.annotation.SuppressLint
import android.content.Intent
import android.content.Intent.CATEGORY_BROWSABLE
import android.net.Uri
import android.os.PersistableBundle
import android.widget.Button
import android.widget.TextView
import androidx.core.view.isVisible
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle

object ListQuestions{
    val questions: List<Question>
        get() = listOf(
            Question("Czy Programowanie urządzeń moblilnych to najlepszy przedmiot?",true),
            Question("Czy słońce to planeta?",false),
            Question("Czy astronauci skakali po Księżycu?",true),
            Question("Czy amper to jednostka natężenia prądu?",true),
            Question("Czy atom jest mniejszy od protonu?",false),
            Question("Czy wszystkie kąty w kwadracie są proste?",true),
            Question("Czy Zero absolutne to najniższa z osiągniętych dotąd temperatur?",false),
            Question("Czy metr w układzie SI jest oznaczony literą 'm'?",true),
            Question("Czy kilogram ma 1000 gram?",true),
            Question("Czy Księżyc to satelita Ziemi?",true)
        )
}

class Question(
    val question: String,
    val correctAnswer: Boolean
)

class MainActivity : AppCompatActivity(){

    private var position = 0
    private var answerCount = 0
    private var points = 0
    private var cheatCount = 0

    private val questions: List<Question> = ListQuestions.questions
    private var question = questions[position]

    private val urlGoogle = "https://www.google.com/search?q="
    private val questionText: TextView by lazy { findViewById(R.id.textQuestion) }

    override fun onCreate(savedInstanceState: Bundle?){
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        if (savedInstanceState != null)
        {
            position = savedInstanceState.getInt("position")
            points = savedInstanceState.getInt("points")
            cheatCount = savedInstanceState.getInt("cheatCount")
            answerCount = savedInstanceState.getInt("answerCount")

        }
        setQuestion()
    }

    override fun onSaveInstanceState(outState: Bundle, outPersistentState: PersistableBundle) {
        outState.putInt("points", points)
        outState.putInt("currentPosition", position)
        outState.putInt("cheats", cheatCount)
        outState.putInt("rightAns", answerCount)
        super.onSaveInstanceState(outState, outPersistentState)
    }

    private fun setQuestion(){
        if(position < questions.size)
        {
            question = questions[position]
            questionText.text = question.question
        }
    }

    @SuppressLint("SetTextI18n")
    private fun showResults(){
        val trueBut: Button by lazy { findViewById(R.id.button)}
        trueBut.isVisible = false
        val falseBut: Button by lazy { findViewById(R.id.button1)}
        falseBut.isVisible = false
        val cheatBut: Button by lazy { findViewById(R.id.button2)}
        cheatBut.isVisible = false
        questionText.text = "Wynik: $points\n Poprawne odpowiedzi: $answerCount\n Oszustwa: $cheatCount"
    }

    fun buttonClickAnswer(view: android.view.View) {
        if(question.correctAnswer == (view as Button).text.toString().toBoolean())
        {
            points += 10
            println(points)
            answerCount++
        }
        position++
        if(position < questions.size) {

            setQuestion()
        }
        else showResults()
    }

    fun buttonClickCheat(view: android.view.View) {
        val intent = Intent(this, CheatActivity::class.java).apply {}
        startActivity(intent)
        points -= 15
        cheatCount++
        if(position < questions.size)
        {
            position++
            setQuestion()
        }
        else showResults()
    }

    @SuppressLint("QueryPermissionsNeeded")
    fun clickQuestion(view: android.view.View) {
        val urlSearch = urlGoogle + question.question.replace(" ", "+")
        val intentSearch = Intent(Intent.ACTION_VIEW, Uri.parse(urlSearch)).apply {
            addCategory(CATEGORY_BROWSABLE)
        }

        if (intent.resolveActivity(packageManager) != null)
        {
            startActivity(intentSearch)
        }
    }

}